main/lsp/ErrorReporter.cc

```diff
@@ -30,23 +30,20 @@ void ErrorReporter::beginEpoch(u4 epoch, vector<unique_ptr<Timer>> diagnosticLat
     epochTimers[epoch] = EpochTimers{move(firstDiagnosticLatencyTimers), move(diagnosticLatencyTimers)};
 }
 
-void ErrorReporter::endEpoch(u4 epoch) {
-    epochTimers.erase(epoch);
-}
-
-void ErrorReporter::cancelEpoch(u4 epoch) {
+void ErrorReporter::endEpoch(u4 epoch, bool committed) {
     auto it = epochTimers.find(epoch);
     ENFORCE(it != epochTimers.end());
+    if (committed) {
+        epochTimers.erase(it);
```

Comment:
? If `!committed` you have to erase the epoch timers as well. Otherwise they will remain in memory forever.